src/views/virtualmachines/migrate/utils.ts

```diff
@@ -0,0 +1,158 @@
+import { VirtualMachineModel } from 'src/views/dashboard-extensions/utils';
+
+import DataVolumeModel from '@kubevirt-ui/kubevirt-api/console/models/DataVolumeModel';
+import {
+  V1beta1DataVolume,
+  V1beta1StorageSpecVolumeModeEnum,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { MAX_NAME_LENGTH } from '@kubevirt-utils/components/SSHSecretModal/utils/constants';
+import { getName, getNamespace } from '@kubevirt-utils/resources/shared';
+import { getDataVolumeTemplates, getVolumes } from '@kubevirt-utils/resources/vm';
+import { getRandomChars } from '@kubevirt-utils/utils/utils';
+import { k8sCreate, k8sDelete, k8sPatch, Patch } from '@openshift-console/dynamic-plugin-sdk';
+
+const getBlankDataVolume = (
+  originName: string,
+  namespace: string,
+  storageClassName: string,
+  storage: string,
+): V1beta1DataVolume => ({
+  apiVersion: `${DataVolumeModel.apiGroup}/${DataVolumeModel.apiVersion}`,
+  kind: DataVolumeModel.kind,
+  metadata: {
+    name: `clone-${getRandomChars()}-${originName}`.substring(0, MAX_NAME_LENGTH),
```

Comment:
i think clone-origin-getrandom make more sense as random should be in the end

Reply:
and if you are worried beacuse of the substring will cut random so like this:
```
const rnadomChars = getRandomChars();
`clone-${originName.substring(0, MAX_NAME_LENGTH - 6 - randomChars.length)}-${randomChars}`
````